tools/icon_recognition/publish.py

```
@@ -69,15 +74,32 @@ def sync_published_images(
+        if not isinstance(fluid_icon_id, str) or not fluid_icon_id:
             continue
         fluid_source = source_by_icon_id.get(fluid_icon_id)
         if fluid_source is None:
             raise ValueError(f"item.json 找不到流体图标对应物品: {fluid_icon_id}")
+        referenced_icon_ids.add(fluid_icon_id)
         expected_images.add(
             Path(str(fluid_source["rarity"])) / f"{fluid_icon_id}.png"
         )
     asset_image_root.mkdir(parents=True, exist_ok=True)
+
+    # 黑名单物品可能与正常物品共用图标；只有当前 catalog 完全不再引用时才能删除。
```

Ruff (0.16.3)

[warning] Comment contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF003)